infrastructure/smart-contracts/contracts/upgrades/node-license/NodeLicense8.sol

```diff
@@ -445,6 +447,7 @@ contract NodeLicense8 is ERC721EnumerableUpgradeable, AccessControlUpgradeable
             // Calculate the referral reward
             referralReward = _finalPrice * referralRewardPercentage / 100;
             emit ReferralReward(msg.sender, promoCode.recipient, referralReward);
+            emit ReferralRewardV2(msg.sender, promoCode.recipient, _currency, _promoCode, referralReward);
```

Comment:
We cant emit both events every time, since we already ahve emitted `ReferralReward` for ETH based referrals.
We should now emit either one depending if its ETH or any token.

Let me know if that makes sense

Reply:
Fixed!